public/js/index.js

```diff
@@ -28,302 +27,292 @@ let isAuthReady = false;
 // event listener setup
 formElement.addEventListener('submit', addLesson);
 modalLessonClose.addEventListener('click', handleCloseLessonModal);
+modalLessonCloseWithESCKey.addEventListener('keydown', handleCloseLessonModalWithESCKey);
 clearBtn.addEventListener('click', handleClear);
-overlay.addEventListener('click', handleCloseLessonModal);
 
 if (!isAuthReady) {
-	appContainer.setAttribute('hidden', '');
-	profileElement.setAttribute('hidden', '');
-	preAuthContainer.removeAttribute('hidden');
-	overlay.setAttribute('hidden', '');
-	modal.setAttribute('hidden', '');
+  appContainer.setAttribute('hidden', '');
+  profileElement.setAttribute('hidden', '');
+  preAuthContainer.removeAttribute('hidden');
+  overlay.setAttribute('hidden', '');
+  modal.setAttribute('hidden', '');
 }
 
 firebase.auth().onAuthStateChanged((user) => {
-	if (user) {
-		if (!isAuthReady) {
-			isAuthReady = true;
-			init(user);
-		}
-	} else {
-		appContainer.setAttribute('hidden', '');
-		profileElement.setAttribute('hidden', '');
-		preAuthContainer.removeAttribute('hidden');
-		overlay.removeAttribute('hidden');
-		modal.removeAttribute('hidden');
-	}
+  if (user) {
+    if (!isAuthReady) {
+      isAuthReady = true;
+      init(user);
+    }
+  } else {
+    appContainer.setAttribute('hidden', '');
+    profileElement.setAttribute('hidden', '');
+    preAuthContainer.removeAttribute('hidden');
+    overlay.removeAttribute('hidden');
+    modal.removeAttribute('hidden');
+  }
 });
 
 function handleClear(e) {
-	quill.root.innerHTML = '';
-	lessonInput.value = '';
-	clearBtn.setAttribute('hidden', '');
-	submitLessonElement.textContent = 'ADD LESSON';
+  quill.root.innerHTML = '';
+  lessonInput.value = '';
+  clearBtn.setAttribute('hidden', '');
+  submitLessonElement.textContent = 'ADD LESSON';
 }
 
 function handleCloseLessonModal() {
-	modalLessonTitle.innerHTML = '';
-	modalLessonContent.innerHTML = '';
+  modalLessonTitle.innerHTML = '';
+  modalLessonContent.innerHTML = '';
+  modalLesson.setAttribute('hidden', '');
+  overlay.setAttribute('hidden', '');
+  overlay.classList.remove('dark');
+}
 
-	modalLesson.setAttribute('hidden', '');
-	overlay.setAttribute('hidden', '');
-	overlay.classList.remove('dark');
+function handleCloseLessonModalWithESCKey(e) {
+  let keyCode = e.keyCode
+  if(keyCode == 27){
```

Comment:
I would recommend using the strict equality operator in JS i.e. `===` rather than standard equality `==`

Sometimes funky things happen (this is actually called Type coercion) like this: 
```
console.log('1' ==  1); // true
```